understory_box_tree/src/tree.rs

```diff
@@ -337,6 +346,26 @@ impl<B: Backend<f64>> Tree<B> {
             .map(|node| node.world.world_bounds)
     }
 
+    /// Return the viewport bounds for a live node.
+    ///
+    /// This returns the unclipped world bounds intersected with the node's
+    /// effective clip (which may include ancestor clips based on [`ClipBehavior`]).
+    /// This represents the actual visible area of the node. Returns `None` for stale
+    /// identifiers.
+    pub fn viewport_bounds(&self, id: NodeId) -> Option<Rect> {
+        if !self.is_alive(id) {
+            return None;
+        }
+        let node = self.nodes.get(id.idx())?.as_ref()?;
+        let unclipped_bounds =
+            transform_rect_bbox(node.world.world_transform, node.local.local_bounds);
+
+        match node.world.world_clip {
+            Some(clip) => Some(unclipped_bounds.intersect(clip)),
+            None => Some(unclipped_bounds),
+        }
```

Comment:
Hmm, this looks very similar to the `world_bounds` calculation. Are there cases where it differs?

The transform is calculated similarly:

https://github.com/endoli/understory/blob/b401400b9b8aa6ed8bd3f5ba55f90411c958788a/understory_box_tree/src/tree.rs#L687-L688

And I believe the same value of `WorldNode::world_clip` is calculated here

https://github.com/endoli/understory/blob/b401400b9b8aa6ed8bd3f5ba55f90411c958788a/understory_box_tree/src/tree.rs#L703-L707

Calling out to `self.world_bounds(id)` also lets tests pass.

Reply:
Ah yes looks like it. I assumed the world bounds wasn't respecting the clips